src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php

```diff
@@ -53,6 +53,9 @@ public function get(RequestConfiguration $requestConfiguration, RepositoryInterf
             $resources->setMaxPerPage($requestConfiguration->getPaginationMaxPerPage());
             $resources->setCurrentPage($request->query->get('page', 1));
 
+            // This prevents Pagerfanta from querying database from a template
+            $resources->getCurrentPageResults();
```

Comment:
It will prevent from querying database only for resources without grid definition.